src/http_protocol.cpp

```diff
@@ -731,4 +749,130 @@ std::string http_response_t::generic(unsigned code,
+            return shared_cors;
+        } else {
+            return shared_blocked_by_cors;
+        }
+    };
+}
+
+// make a short circuiter that will respond to health checks and OPTIONS request
+// Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+shortcircuiter_t<http_request_t> make_shortcircuiter(const std::string& health_check_path,
+                                                     const uint8_t& verb_mask,
+                                                     const std::string& allowed_origins,
+                                                     const std::string& allowed_headers,
+                                                     const int max_age){
+
+  std::string allowed_verbs = get_allowed_methods_string(verb_mask);
+
+  auto cors_handler = handle_cors(allowed_verbs, allowed_origins, allowed_headers, max_age);
+
+  // Pre-created shared response objects
+  http_response_t health_check_response(200, "OK");
+  std::string health_check_str = health_check_response.to_string();
+  std::shared_ptr<zmq::message_t> shared_health_check = std::make_shared<zmq::message_t>(health_check_str.size(), health_check_str.c_str());
+
+  http_response_t method_not_allowed_response(405, "Method Not Allowed");
+  std::string method_not_allowed_str = method_not_allowed_response.to_string();
+  std::shared_ptr<zmq::message_t> shared_method_not_allowed = std::make_shared<zmq::message_t>(method_not_allowed_str.size(), method_not_allowed_str.c_str());
+
+  http_response_t options_response(200, "OK");
+  options_response.headers.insert({"Allow", allowed_verbs});
+  std::string options_str = options_response.to_string();
+  std::shared_ptr<zmq::message_t> shared_options = std::make_shared<zmq::message_t>(options_str.size(), options_str.c_str());
+
+  http_response_t cors_response(200, "OK");
+  cors_response.headers.insert({"Access-Control-Allow-Origin", allowed_origins});
+  cors_response.headers.insert({"Access-Control-Allow-Methods", allowed_verbs});
+  cors_response.headers.insert({"Access-Control-Allow-Headers", allowed_headers});
+  cors_response.headers.insert({"Access-Control-Max-Age", std::to_string(max_age)});
+  std::string cors_str = cors_response.to_string();
+  std::shared_ptr<zmq::message_t> shared_cors = std::make_shared<zmq::message_t>(cors_str.size(), cors_str.c_str());
+
+  return [=](const http_request_t& request) {
+    if (request.path == health_check_path) {
+      return shared_health_check;
+    } 
+    else if (!(verb_mask & static_cast<uint8_t>(request.method))) {
+      return shared_method_not_allowed;
+    } 
+    else if (request.method == method_t::OPTIONS) {
+      if(request.headers.find("Access-Control-Request-Method") != request.headers.end()) {
+        return cors_handler(request);
+      }else {
+        return shared_options;
+      }
+    } 
+    else {
+      return std::shared_ptr<zmq::message_t>();
+    }
+  };
+}
+// namespace prime_server
+}
```

Comment:
there is a mistake here somewhere :smile:

Reply:
hahaha